src/Service/GoogleGroupsService.php

```diff
@@ -224,4 +225,19 @@ public function removeMember(string $groupEmail, string $email)
             ];
         }
     }
+
+    private function getMfaGroupUserEmails(): array
+    {
+        $users = $this->getMembers(self::MFA_EXCEPTION_GROUP);
+        $emails = [];
+        foreach ($users as $user) {
+            $emails[] = $user->email;
+        }
+        return $emails;
+    }
+
+    public function isMfaGroupUser($email): bool
+    {
+        return in_array($email, $this->getMfaGroupUserEmails());
```

Comment:
Instead of retrieving all members in the group and looking for the one user, I think it would be better to look at all of the user's groups to see if they are a member of the exception group.

I think the number of new users being onboarded and belonging to the 2fa exception group could grow as the AoU program grows, while the maximum number of sites that a single user belongs to would be fairly static.